creators_custom/RoutesCreatorFenix.py

```diff
@@ -0,0 +1,508 @@
+            sim_stop = self.normalize_stop_name(sim_stop)
+            sys.stderr.write("SIM Stop: '" + sim_stop + "'\n")
+        print
+        return None
+
+    def get_first_stop(self):
+            if len(self.stops) > 0:
+                return self.stops[0]
+            else:
+                return None
+
+    def get_first_alt_stop(self):
+        if self.fr is not None:
+            return self.fr
+        else:
+            return "???"
+
+    def has_proper_master(self):
+        return self.master is not None and len(self.master.routes) > 1
+
+    def add_shape(self, refresh=False):
+        if self.shape is not None and not refresh:
+            return
+
+        self.shape = []
+
+        ways = []
+        osm_api = OsmApi()
+        for member in osm_api.RelationGet(self.id)['member']:
+            if member["type"] == "way" and not member["role"].startswith("platform"):
+                ways.append(member["ref"])
+
+        nodes = []
+        ways_info = osm_api.WaysGet(ways)
+        for way in ways:
+            if len(nodes) == 0:
+                nodes.extend(ways_info[way]["nd"])
+            elif nodes[-1] == ways_info[way]["nd"][0]:
+                del nodes[-1]
+                nodes.extend(ways_info[way]["nd"])
+            elif nodes[-1] == ways_info[way]["nd"][-1]:
+                del nodes[-1]
+                nodes.extend(reversed(ways_info[way]["nd"]))
+            elif nodes[0] == ways_info[way]["nd"][0]:
+                del nodes[0]
+                nodes.reverse()
+                nodes.extend(ways_info[way]["nd"])
+            elif nodes[0] == ways_info[way]["nd"][-1]:
+                del nodes[0]
+                nodes.reverse()
+                nodes.extend(reversed(ways_info[way]["nd"]))
+            else:
+                sys.stderr.write("Route has non-matching ways: " + str(self) + "\n")
+                sys.stderr.write("  Problem at: http://www.openstreetmap.org/way/" + str(ways_info[way]["id"]) + "\n")
+                break
+
+        # limit the number of nodes per API request to prevent 414
+        size = 100
+        for nodes_sublist in [nodes[i:i + size] for i in range(0, len(nodes), size)]:
+            nodes_info = osm_api.NodesGet(nodes_sublist)
+            for node in nodes_sublist:
+                self.shape.append({'lat': nodes_info[node]["lat"], 'lon': nodes_info[node]["lon"]})
+
+    def print_shape_for_leaflet(self):
+        print "var shape = [",
+        i = 0
+        for node in self.shape:
+            print "new L.LatLng(" + str(node["lat"]) + ", " + str(node["lon"]) + ")",
+            if i != len(self.shape) - 1:
+                print ",",
+            i += 1
+        print "];"
+        i = 0
+        for node in self.shape:
+            print "L.marker([" + str(node["lat"]) + ", " + str(node["lon"]) + "]).addTo(map)"
+            print "    .bindPopup(\"" + str(i) + "\").openPopup();"
+            i += 1
+
+    def normalize_stop_name(old_name):
```

Comment:
I think this can again be a static method, otherwise it needs `self` as first parameter.